src/ConnectedMode/Migration/FileProviders/XmlFileProvider.cs

```diff
@@ -36,7 +34,7 @@ namespace SonarLint.VisualStudio.ConnectedMode.Migration.FileProviders
     [PartCreationPolicy(CreationPolicy.NonShared)]
     internal partial class XmlFileProvider : IFileProvider
     {
-        private readonly IServiceProvider serviceProvider;
+        private readonly ISolutionInfoProvider solutionInfoProvider;
         private readonly IRoslynProjectProvider projectProvider;
         private readonly ILogger logger;    
         private readonly IThreadHandling threadHandling;
```

Comment:
This class still needs `IThreadHandling` to make sure the file operations are done on a background thread.